moveit_planners/chomp/chomp_motion_planner/src/chomp_planner.cpp

```diff
@@ -156,6 +156,33 @@ bool ChompPlanner::solve(const planning_scene::PlanningSceneConstPtr& planning_s
 
   ros::WallTime create_time = ros::WallTime::now();
 
+  /// block for replanning (recovery behaviour) if collision free optimized solution not found
+  int replan_cnt = 0;
+  bool replan_flag = false;
+  double org_learning_rate = 0.04, org_ridge_factor = 0.0, org_planning_time_limit = 10;
+  int org_max_iterations = 200;
+RePlan:
+{
+  if (replan_flag)
+  {
+    params.learning_rate_ *= 0.02;     // increase learning rate in hope to find obstacles (possibly faster convergence)
+    params.ridge_factor_ += 0.002;     // increase ridge factor to avoid obstacles
+    params.planning_time_limit_ += 5;  // add 5 additional secs in hope to find a solution
+    params.max_iterations_ += 50;      // increase maximum iterations
```

Comment:
how did you choose these values? can you add a plot to this PR showing convergence given these parameters?

Reply:
these values are currently based on rough empirical analysis and intuition.